File: app/src/main/java/de/mchllngr/quickero/module/about/AboutActivity.java
package de.mchllngr.quickero.module.about;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import de.mchllngr.quickero.BuildConfig;
import de.mchllngr.quickero.R;
import de.mchllngr.quickero.base.BaseActivity;
import de.mchllngr.quickero.base.BasePresenter;
import de.mchllngr.quickero.base.BaseView;
import mehdi.sakout.aboutpage.AboutPage;
import mehdi.sakout.aboutpage.Element;

/**
 * {@link Activity} for the AboutPage.
 */
public class AboutActivity extends BaseActivity<BaseView, BasePresenter<BaseView>> implements BaseView {

    private static final String DESCRIPTION = "TODO description"; // TODO set description
    private static final String EMAIL = "quickero@michaellanger.me";
    private static final String GITHUB = "mchllngr/quickero";

    public static void start(Context context) {
        Intent starter = new Intent(context, AboutActivity.class);
        context.startActivity(starter);
    }

    @NonNull
    @Override
    public BasePresenter<BaseView> createPresenter() {
        return new BasePresenter<>();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(buildAboutPage());
    }

    /**
     * Builds the {@link View} of the {@link AboutActivity}.
     */
    private View buildAboutPage() {
        Element versionElement = new Element();
        versionElement.setTitle("Version " + BuildConfig.VERSION_NAME);

        return new AboutPage(this)
                .setImage(R.drawable.ic_launcher_foreground)
                .setDescription(DESCRIPTION)
                .addItem(versionElement)
                .addEmail(EMAIL)
                .addPlayStore(BuildConfig.APPLICATION_ID)
                .addGitHub(GITHUB)
                .create();
    }
}

File: app/src/main/java/de/mchllngr/quickero/module/about/AboutActivity.kt
package de.mchllngr.quickero.module.about

import android.content.Context
import android.content.Intent
import android.os.Bundle
import androidx.annotation.StringRes
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.content.res.AppCompatResources
import androidx.fragment.app.commit
import com.google.android.material.appbar.MaterialToolbar
import com.mikepenz.aboutlibraries.LibsBuilder
import de.mchllngr.quickero.R

class AboutActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_about)

        findViewById<MaterialToolbar>(R.id.toolbar)?.let { toolbar ->
            setSupportActionBar(toolbar)
            toolbar.navigationIcon = AppCompatResources.getDrawable(this, R.drawable.ic_baseline_arrow_back_24)
            toolbar.setNavigationOnClickListener { onBackPressed() }
        }

        val description = buildString {
            append("<b>")
            appendUrl(R.string.url_github, R.string.github)
            appendSeparator()
            appendUrl(R.string.url_play_store, R.string.play_store)
            appendSeparator()
            appendUrl(R.string.url_privacy_policy, R.string.privacy_policy)
            append("</b>")
        }

        supportFragmentManager.commit {
            replace(
                R.id.fragment_container_view,
                LibsBuilder()
                    .withFields(R.string::class.java.fields)
                    .withAboutDescription(description)
                    .supportFragment()
            )
        }
    }

    private fun StringBuilder.appendUrl(
        @StringRes url: Int,
        @StringRes text: Int
    ) {
        append("<a href=\"")
        append(getString(url))
        append("\">")
        append(getString(text))
        append("</a>")
    }

    private fun StringBuilder.appendSeparator() {
        append("&nbsp;&nbsp;|&nbsp;&nbsp;")
    }

    companion object {

        @JvmStatic
        fun start(context: Context) {
            context.startActivity(Intent(context, AboutActivity::class.java))
        }
    }
}
